apis/batch/v1alpha1/taskqueue_types.go

```diff
@@ -0,0 +1,133 @@
+/*
+Copyright 2025.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+type TaskPhase string
+
+const (
+	TaskPhaseInPending  TaskPhase = "InPending"
+	TaskPhaseInProgress TaskPhase = "InProgress"
+	TaskPhaseFailed     TaskPhase = "Failed"
+	TaskPhaseSuccessful TaskPhase = "Successful"
+)
+
+// +k8s:openapi-gen=true
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+
+// TaskQueue is the Schema for the taskqueues API.
+type TaskQueue struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   TaskQueueSpec   `json:"spec,omitempty"`
+	Status TaskQueueStatus `json:"status,omitempty"`
+}
+
+// TaskQueueSpec defines the desired state of TaskQueue.
+type TaskQueueSpec struct {
+	// MaxConcurrentTasks specifies the maximum number of tasks can run concurrently.
+	// Defaults to 20 if not set.
+	// +kubebuilder:default=20
+	// +optional
+	MaxConcurrentTasks int `json:"maxConcurrentTasks,omitempty"`
+
+	// Tasks represents the lists of tasks that this queue is responsible for processing.
+	Tasks []UnitTask `json:"tasks,omitempty"`
+}
+
+// ObjectPhaseRules defines three identification rules of successful phase of the object,
+// progressing phase of the object & failed execution of the object.
+// To specifies any field of the Operation object, the rule must start with the word `self`.
+// Example:
+//
+//	.status.phase -> self.status.phase
+//	.status.observedGeneration -> self.status.observedGeneration
+//
+// The rules can be any valid expression supported by CEL(Common Expression Language).
+// Ref: https://github.com/google/cel-spec
+type ObjectPhaseRules struct {
```

Comment:
Move this struct after the TypedResourceReference definition.  
Just to make it well organized.